### PR TITLE
[hdf5] Update version to 1.14.4.3

### PR DESCRIPTION
**Pull request by hyoklee**

### CODE REVIEW

**Comment by hyoklee** [comment]
@microsoft-github-policy-service agree [company="{The HDF Group}"]

**Comment by JonLiu1993** [comment]
Please run command `./vcpkg x-add-version hdf5` and commit agagin.

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/hdf5/portfile.cmake [hunk 1]
```diff
@@ -1,16 +1,15 @@
 # highfive should be updated together with hdf5
 
-string(REPLACE "." "_" hdf5_ref "hdf5-${VERSION}")
+string(REPLACE "." "." hdf5_ref "hdf5_${VERSION}")
```
Comment (JonLiu1993) [inline ports/hdf5/portfile.cmake:3]: Is there any difference between before and after replacing this line?

Comment (hyoklee) [reply ports/hdf5/portfile.cmake:3]: @JonLiu1993 , no. I left it just in case HDF5 release changes version scheme again in the future release.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
@hyoklee, thanks for your PR, when I tested the feature with command "./vcpkg install hdf5[*]:x64-windows", I get this error, please take a look:
````
CMake Error at config/cmake/HDFUseFortran.cmake:16 (enable_language):
  No CMAKE_Fortran_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "FC" or the CMake cache entry CMAKE_Fortran_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  config/cmake/HDF5UseFortran.cmake:17 (include)
  CMakeLists.txt:1076 (include)
````

**Comment by Neumann-A** [comment]
@JonLiu1993 remove the fortran feature from your install command..... you don't have a fortran compiler setup

**Comment by hyoklee** [comment]
@JonLiu1993 thanks for testing all options. I could duplicate your error. 
FYI, the same error occurs in the current hdf5 vcpkg port 1.14.2.
Do you want me to remove fortran option completely as suggested by @Neumann-A 
or give a warning to user via message?

**Comment by Neumann-A** [comment]
> Do you want me to remove fortran option completely as suggested by @Neumann-A

No don't remove it.I just that that @JonLiu1993 test should remove it from the install command. It works if you have a Fortran compiler setup....

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature [`cpp,map,parallel,szip,tools,utils,zlib`] tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by hyoklee** [comment]
Thank you, @vicroms,  for merging this PR!

**Pull request merged**